railties/lib/rails/application/configuration.rb

```diff
@@ -303,6 +307,10 @@ def load_defaults(target_version)
               "Referrer-Policy" => "strict-origin-when-cross-origin"
             }
             action_dispatch.debug_exception_log_level = :error
+
+            if Rails.env.test?
+              action_dispatch.show_exceptions = :rescuable
+            end
```

Comment:
I'm :+1: on adding the section to `configuring` because I think all configrations should definitely be documented, but I don't understand the premise of this PR asserting that any default changed. It looks like this line is changing the default but that was not the case before.

If the assertion is that changes to the environment files constitutes "new defaults" then I'm not sure I agree with that. A user has to actively accept that new configuration and should be inspecting the diff of the environment files for changes when running `app:update`.

Reply:
@skipkayhil I think you're right -- I'm conflating the default generated by the `config/environments/test.rb` template with whatever Rails defaults to when that override is missing.

> I don't understand the premise of this PR asserting that any default changed

When I generate an app with `rails new` then make this change:

```diff
 # Render exception templates for rescuable exceptions and raise for other exceptions.
-  config.action_dispatch.show_exceptions = :rescuable
```

The behavior appears to be as if I'd set `:all`.

Maybe something is happening somewhere in my `rails new` configuration. When I try and reproduce with the bug report template, things behave as if `show_exceptions` were set to `false` or `:none`:

```ruby
# frozen_string_literal: true

require "bundler/inline"

gemfile(true) do
  source "https://rubygems.org"

  git_source(:github) { |repo| "https://github.com/#{repo}.git" }

  # Activate the gem you are reporting the issue against.
  gem "rails", "~> 7.1.0"
end

require "action_controller/railtie"

class TestApp < Rails::Application
  config.load_defaults 7.1
  config.root = __dir__
  config.hosts << "www.example.com"
  config.session_store :cookie_store, key: "cookie_store_key"
  config.secret_key_base = "secret_key_base"
  config.consider_all_requests_local = true

  config.logger = Logger.new($stdout)
  Rails.logger  = config.logger
  ActionDispatch.test_app = self

  routes.draw do
    get "/" => "test#index"
  end
end

class TestController < ActionController::Base
  def index
    # raise StandardError, "Not rescuable"
    render plain: "Hello"
  end
end

require "rails/test_help"

class BugTest < ActionDispatch::IntegrationTest
  def test_returns_success
    get "/"

    assert_response :ok
  end
end
```

I'm happy to revert the changes that mention "default" values, but I'd also like to troubleshoot why removing that line from a new application wouldn't exhibit the same behavior as the Bug Report Template app.

Reply:
@skipkayhil by default, the `action_dispatch` Railtie sets the value to `:all`:

https://github.com/rails/rails/blame/main/actionpack/lib/action_dispatch/railtie.rb#L12

Prior to that change, the Railtie default was `true` (equivalent to `:all`):

https://github.com/rails/rails/blame/1173d2ce0489f7968f549a3098a39aaf6710586c/actionpack/lib/action_dispatch/railtie.rb#L12

To clarify, this PR's intent was to highlight the **test** environment defaults. I had mistaken this for a **test**-only configuration that didn't have a **production** side impact. That's my mistake.

With that in mind, I'm curious to how proceed. End-users are not at all affected by the changes made in Rails 7.1. Despite that, the development experience changed dramatically by defaulting to `:rescuable`. I think that's something worth highlighting with documentation.

Reply:
> @skipkayhil by default, the `action_dispatch` Railtie sets the value to `:all`:
> 
> https://github.com/rails/rails/blame/main/actionpack/lib/action_dispatch/railtie.rb#L12
> 
> Prior to that change, the Railtie default was `true` (equivalent to `:all`):
> 
> https://github.com/rails/rails/blame/1173d2ce0489f7968f549a3098a39aaf6710586c/actionpack/lib/action_dispatch/railtie.rb#L12
> 
> To clarify, this PR's intent was to highlight the **test** environment defaults. I had mistaken this for a **test**-only configuration that didn't have a **production** side impact. That's my mistake.
> 
> With that in mind, I'm curious to how proceed. End-users are not at all affected by the changes made in Rails 7.1. Despite that, the development experience changed dramatically by defaulting to `:rescuable`. I think that's something worth highlighting with documentation.

I definitely see value in documenting the available options, especially since they changed between 7.0 and 7.1. It looks like the only mention is currently in the release notes, but nowhere is it explained what the three options do. I think that's probably a good place to start.

Reply:
Oh and I definitely think a section in the upgrade guide may be nice as well... I think the "default" wording is just not quite right, something like:

```md
### New ways to handle exceptions in Integration Tests

In Rails 7.0, show_exceptions could be enabled to .... and show_exceptions is set to true in config/environments/test.rb for new applications.

Rails 7.1 reworks the options to add a new way... New apps will set show_exceptions to :rescuable in config/enviornments/test.rb and upgrading apps can either use the new `:rescuable` or simply switch their old `true`/`false` to `:all`/`:none`...
```